src/main/java/uk/gov/companieshouse/overseasentitiesapi/utils/OEPublicDataRetrievalHelper.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.companieshouse.overseasentitiesapi.utils;
+
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.handler.exception.URIValidationException;
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.overseasentitiesapi.client.ApiClientService;
+import uk.gov.companieshouse.overseasentitiesapi.exception.ServiceException;
+
+import java.io.IOException;
+
+@Component
+public class OEPublicDataRetrievalHelper {
+    private final ApiClientService apiClientService;
+
+    public OEPublicDataRetrievalHelper(ApiClientService apiClientService) {
+        this.apiClientService = apiClientService;
+    }
+
+    public void getOverseasEntityPublicData(String companyReference, String passThroughTokenHeader) throws ServiceException {
+        getCompanyProfile(companyReference, passThroughTokenHeader);
+    }
+    private CompanyProfileApi getCompanyProfile(String companyReference, String passThroughTokenHeader) throws ServiceException {
+        try {
+            CompanyProfileApi companyProfileData = apiClientService
```

Comment:
you joined up two lines for payments above but not here. Maybe revert the payment change?

Reply:
Not sure what you mean, could you please clarify @AndreKramer7654

Reply:
Ok just realised what happened, that payment change was unintentional. Might have been intellij trying to be smart